components/jaggery-core/org.jaggeryjs.jaggery.core/src/main/java/org/jaggeryjs/jaggery/core/manager/ModuleManager.java

```diff
@@ -116,56 +115,57 @@ public JavaScriptModule getModule(String name) throws ScriptException {
     private void initModule(InputStream modulesXML, boolean isCustom) throws ScriptException {
         try {
             Context cx = RhinoEngine.enterGlobalContext();
-            StAXOMBuilder builder = new StAXOMBuilder(modulesXML);
-            OMElement document = builder.getDocumentElement();
 
-            OMNamespace ns = document.getNamespace();
-            if (ns == null || !MODULE_NAMESPACE.equals(document.getNamespace().getNamespaceURI())) {
-                log.warn("A module xml found without the proper namespace");
-                return;
-            }
+            JAXBContext jaxbContext = JAXBContext.newInstance(Module.class);
+            Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
+
+            Module moduleObject = (Module) jaxbUnmarshaller.unmarshal(modulesXML);
 
-            String moduleName = document.getAttributeValue(new QName(null, NAME));
+//            if (moduleObject.getNamespace() == null || !MODULE_NAMESPACE.equals(moduleObject.getNamespace())) {
+//                log.warn("A module xml found without the proper namespace");
+//                return;
+//            }
+            String moduleName = moduleObject.getName();
 
             if (modules.get(moduleName) != null) {
                 log.info("A module with the name : " + moduleName + " already exists and it will be overwritten.");
             }
 
-            String namespace = document.getAttributeValue(new QName(null, NAMESPACE));
-            boolean expose = "true".equalsIgnoreCase(document.getAttributeValue(new QName(null, EXPOSE)));
+            String namespace = moduleObject.getNamespace();
+            boolean expose = ("true".equals(moduleObject.getExpose()));
```

Comment:
Don't do string equals check for the boolean, do parseBoolean and covert string into boolean. Also need to do a null check.